{
  final Throwable cause=exception.getCause();
  if (cause instanceof IOException) {
    return (IOException)cause;
  }
  if (cause instanceof RuntimeException) {
    throw (RuntimeException)cause;
  }
  throw exception;
}
